core/src/main/java/org/elasticsearch/action/admin/indices/recovery/ShardRecoveryResponse.java

```diff
@@ -63,26 +67,37 @@ public void recoveryState(RecoveryState recoveryState) {
      * @return  Recovery state
      */
     @Nullable
-    public RecoveryState recoveryState() {
-        return recoveryState;
+    public List<RecoveryState> recoveryStates() {
+        return recoveryStates;
     }
 
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-        recoveryState.toXContent(builder, params);
+        builder.startArray("recovery_states");
```

Comment:
doesn't this add another field name that wasn't there before? Is this method used? do we need to implement toXContent? I'm looking a the implementation of RecoveryResponse